SQL_module/SubjectForm_Task2/db_manipulation.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+    include "./db_connection.php";
+
+    $table_name = "electives";
+
+    function insertInTable($title, $description, $lecturer){
+        try {
+
+            $connection = db_connection();
+            $sql = "INSERT INTO electives(title,description,lecturer) VALUES (:title, :description, :lecturer)";
+    
+            $preparedSql = $connection->prepare($sql) or die("Failed to prepare sql query.");
+            $preparedSql->bindParam(':title', $title);
+            $preparedSql->bindParam(':description', $description);
+            $preparedSql->bindParam(':lecturer', $lecturer);
+            
+            $preparedSql->execute() or die("Filed to execute sql query.");   
+        }
+        catch(PDOException $error) {
+            echo $error->getMessage();
+        }
+    }
+
+    function printContentOfTable(){
+        
+        global $table_name;
+        try {
+            $connection = db_connection();
+            $sql = "SELECT * FROM $table_name";
+            
+            $result = $connection->prepare($sql) or die("Filed to prepare sql query.");
+            $result->execute() or die("Filed to execute sql query.");
+            while($row = $result->fetch(PDO::FETCH_ASSOC)) {
+                echo "ID: " . $row["id"] . "<br/>Title: " . $row["title"] . "<br/>Description: " . $row["description"] . "<br/>Lecturer: " . $row["lecturer"] . "<br/>Created_at: " . $row["created_at"] . "<br/><br/>";
+            }
+        }
+        catch(PDOException $error) {
+            echo $error->getMessage();
+        }
+    }
+
+    function printSubject($id){
+        global $table_name;
+        try {
+            $connection = db_connection();
+            $sql = "SELECT * FROM $table_name WHERE id = :id;";
+            
+            $result = $connection->prepare($sql) or die("Filed to prepare sql query.");
+            $result->bindParam(':id', $id);
+            $result->execute() or die("Filed to execute sql query.");
+            while($row = $result->fetch(PDO::FETCH_ASSOC)) {
+                echo "Subject ID: " . $row["id"] . "<br/>Title: " . $row["title"] . "<br/>Description: " . $row["description"] . "<br/>Lecturer: " . $row["lecturer"] . "<br/><br/>";
+            }
+        }
+        catch(PDOException $error) {
+            echo $error->getMessage();
```

Comment:
Here you are basically outputting to the user the raw message of the exception. Sometimes raw messages may contain sensitive information that we may not want to share with the end users.